tests/suite/test_policy_ingress_class.py

```diff
@@ -0,0 +1,162 @@
+        policy_info = read_custom_resource(kube_apis.custom_objects, test_namespace, "policies", pol_name)
+        occur = []
+        t_end = time.perf_counter() + 1
+        resp = requests.get(
+            virtual_server_setup.backend_1_url, headers={"host": virtual_server_setup.vs_host},
+        )
+        print(resp.status_code)
+        assert resp.status_code == 200
+        while time.perf_counter() < t_end:
+            resp = requests.get(
+                virtual_server_setup.backend_1_url, headers={"host": virtual_server_setup.vs_host},
+            )
+            occur.append(resp.status_code)
+        delete_policy(kube_apis.custom_objects, pol_name, test_namespace)
+        self.restore_default_vs(kube_apis, virtual_server_setup)
+        assert (
+            policy_info["status"]
+            and policy_info["status"]["reason"] == "AddedOrUpdated"
+            and policy_info["status"]["state"] == "Valid"
+        )
+        assert occur.count(200) <= 1
+
+    @pytest.mark.parametrize("src", [vs_policy_src])
+    def test_policy_matching_ingress_class(
+            self, kube_apis, crd_ingress_controller, virtual_server_setup, test_namespace, src,
+    ):
+        """
+        Test if rate-limiting policy is working with 1 rps
+        """
+        print(f"Create rl policy")
+        pol_name = create_policy_from_yaml(kube_apis.custom_objects, policy_ingress_class_src, test_namespace)
+        print(f"Patch vs with policy: {src}")
+        patch_virtual_server_from_yaml(
+            kube_apis.custom_objects,
+            virtual_server_setup.vs_name,
+            src,
+            virtual_server_setup.namespace,
+        )
+
+        wait_before_test()
+        policy_info = read_custom_resource(kube_apis.custom_objects, test_namespace, "policies", pol_name)
+        occur = []
+        t_end = time.perf_counter() + 1
+        resp = requests.get(
+            virtual_server_setup.backend_1_url, headers={"host": virtual_server_setup.vs_host},
+        )
+        print(resp.status_code)
+        assert resp.status_code == 200
+        while time.perf_counter() < t_end:
+            resp = requests.get(
+                virtual_server_setup.backend_1_url, headers={"host": virtual_server_setup.vs_host},
+            )
+            occur.append(resp.status_code)
+        delete_policy(kube_apis.custom_objects, pol_name, test_namespace)
+        self.restore_default_vs(kube_apis, virtual_server_setup)
+        assert (
+                policy_info["status"]
+                and policy_info["status"]["reason"] == "AddedOrUpdated"
+                and policy_info["status"]["state"] == "Valid"
+        )
+        assert occur.count(200) <= 1
+
+    @pytest.mark.parametrize("src", [vs_policy_src])
+    def test_policy_non_matching_ingress_class(
+            self, kube_apis, crd_ingress_controller, virtual_server_setup, test_namespace, src,
+    ):
+        """
+        Test if rate-limiting policy is working with 1 rps
```

Comment:
it looks like this test case wasn't fully implemented.
the expected behavior here is for NGINX to return 500 and for the IC to report   
 Reason:   AddedOrUpdatedWithWarning
  State:    Warning
 and the message that the policy is missing or invalid 
 in the VS status and an event similar to this one
 ```
   Warning  AddedOrUpdatedWithWarning  55s   nginx-ingress-controller  Configuration for default/virtual-server was added or updated ; with warning(s): Policy default/rate-limit-primary is missing or invalid
```

Reply:
updated